src/Traveler.js

```diff
@@ -1,14 +1,78 @@
 class Traveler {
```

Comment:
Nice, the traveler class makes a lot more sense this way. Looks lie it was a BIG job!